src/stats/dendrogram.jl

```diff
@@ -0,0 +1,123 @@
+"""
+    dendrogram(x, y; kwargs...)
+
+Draw a [dendrogram](https://en.wikipedia.org/wiki/Dendrogram),
+with leaf nodes specified by `x` and `y` coordinates,
+and parent nodes identified by `merges`.
+
+# Arguments
+- `x`: x positions of leaf nodes
+- `y`: y positions of leaf nodes (default = 0)
+# Keywords
+- `merges`: specifies connections between nodes (see below)
+- `treestyle`: one of `:??`, `:??` 
+"""
+@recipe(Dendrogram, x, y) do scene
+    Theme(
+        weights = automatic,
+        color = theme(scene, :patchcolor),
+        colormap = theme(scene, :colormap),
+        colorrange = automatic,
+        orientation = :vertical,
+        strokecolor = theme(scene, :patchstrokecolor),
+        strokewidth = theme(scene, :patchstrokewidth),
+        cycle = [:color => :patchcolor],
+        inspectable = theme(scene, :inspectable)
+    )
```

Comment:
`theme` is a bit misleading, it only provides the current default theme IIRC.  Better to use `inherit` here.

```suggestion
    Theme(
        weights = automatic,
        color = Makie.inherit(scene, :color, :black),
        colormap = Makie.inherit(scene, :colormap, :viridis),
        colorrange = automatic,
        orientation = :vertical,
        strokecolor = Makie.inherit(scene, :strokecolor, :black),
        strokewidth = Makie.inherit(scene, :strokewidth, 1.0),
        cycle = [:color => :patchcolor],
        inspectable = Makie.inherit(scene, :inspectable)
    )
```